File: ios/metric/HeartBeatMeasurer.m
//
//  HeartBeatMeasurer.m
//  sbp
//
//  Created by Alex on 01/09/2019.
//  Copyright © 2019 Facebook. All rights reserved.
//

#import <Foundation/Foundation.h>
#import "React/RCTBridgeModule.h"

@interface RCT_EXTERN_MODULE(HeartBeatMeasurer, NSObject)

RCT_EXTERN_METHOD(startHeartRateCalculation: (RCTResponseSenderBlock)callback)
RCT_EXTERN_METHOD(getHeartRate: (RCTResponseSenderBlock)callback)

@end

File: ios/metric/HeartBeatMeasurer.swift
import Foundation
import CoreBluetooth

@objc(HeartBeatMeasurer) class HeartBeatMeasurer: NSObject {
  
  private var miBand: CBPeripheral!
  private var hrControlPointCharacteristic: CBCharacteristic?
  private var hrMeasurementCharacteristic: CBCharacteristic?
  private var sensorService: CBService?
  private var sensorCharacteristic: CBCharacteristic?
  
  @objc static func requiresMainQueueSetup() -> Bool {
    return true
  }
  
  @objc func startHeartRateCalculation(_ callback: RCTResponseSenderBlock) {
    print("startHeartRateCalculation called")
    callback([NSNull(), -1])
  }
  
  @objc func getHeartRate(_ callback: RCTResponseSenderBlock){
    callback([NSNull(), -1])
  }
  
  func onHeartRateReceived(_ heartRate: Int) {
    print("BPM: \(heartRate)")
  }
  
}





